### PR TITLE
fix: verify trailers

### PR DESCRIPTION
**Pull request by ronag**
Fixes: https://github.com/nodejs/undici/issues/432

### CODE REVIEW

File: lib/core/errors.js [hunk 1]
```diff
@@ -88,6 +88,16 @@ class ContentLengthMismatchError extends UndiciError {
   }
 }
 
+class TrailerMismatchError extends UndiciError {
+  constructor (message) {
+    super(message)
+    Error.captureStackTrace(this, TrailerMismatchError)
```
Comment (szmarczak) [inline lib/core/errors.js:94]: Is this really needed? Why?

Comment (ronag) [reply lib/core/errors.js:94]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error

Comment (ronag) [reply lib/core/errors.js:94]: https://nodejs.org/api/errors.html#errors_error_capturestacktrace_targetobject_constructoropt

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/core/errors.js (hunk 1 above)
Comment (szmarczak) [reply lib/core/errors.js:94]: But `UndiciError` extends `Error`, so isn't `.stack` generated automatically?

Comment (ronag) [reply lib/core/errors.js:94]: I'm actually unsure why this is necessary but given the best practices I've been able to read it seems to be the correct way...

Comment (ronag) [reply lib/core/errors.js:94]: See the example from MDN:

```js
class CustomError extends Error {
  constructor(foo = 'bar', ...params) {
    // Pass remaining arguments (including vendor specific ones) to parent constructor
    super(...params)

    // Maintains proper stack trace for where our error was thrown (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, CustomError)
    }

    this.name = 'CustomError'
    // Custom debugging information
    this.foo = foo
    this.date = new Date()
  }
}
```

**Pull request merged**